README.md

```diff
@@ -55,12 +55,25 @@ To run Cirrus tasks, simply switch to a directory where the `.cirrus.yml` is loc
 cirrus run
 ```
 
-It is also possible to run a task by name:
+By default, working directory will be `rsync`ed into a container while respecting `.gitignore`
+configuration. This makes sure Cirrus Tasks are executed from a clean state only with source code
+changes.
+
+There is an option to directly mount working directory as is with `--dirty` flag which
```

Comment:
We should probably explain "read-only" in this paragraph better, since it makes little sense with `--dirty`, which is actually a mode in which writes are done against the original project directory.

How about replacing this paragraph with:

>In case `rsync`-ing the whole working directory is too costly, you can pass a `--dirty` flag which will result in all operations being done against the actual working directory (and not it's copy):

>```shell script
>cirrus run --dirty Lint
>```

>Since most linters and code-analysis tools are read-only by their nature — greater speed is achieved at the cost of additional isolation.

Reply:
How about:

>  (and not it's `rsync`ed copy)

> Since most linters and code-analysis tools are read-only by their nature there is no need in extra precautions and the potentially costly `rsync`-ing can be safely avoided.

Reply:
OK.

Reply:
See 5b6d7bc